packages/nimble-components/src/table/specs/table-column-specs/table-column-menu-button.md

```diff
@@ -0,0 +1,205 @@
+    </nimble-table-column-menu-button>
+
+    <nimble-menu slot="software-version-menu">
+        <nimble-menu-item>1.0.0</nimble-menu-item>
+        <nimble-menu-item>1.5.0</nimble-menu-item>
+        <nimble-menu-item>2.0.0</nimble-menu-item>
+    </nimble-menu>
+</nimble-table>
+```
+
+### API
+
+#### Column Component
+
+_Component Name_
+
+-   `nimble-table-column-menu-button`
+
+_Props/Attrs_
+
+-   `field-name`: string - The name of the field in each record that contains the string that will be displayed in the menu button
+-   `menu-slot`: string - The name of the slot within the `nimble-table` instance where the menu associated with the column's menu button will be provided
+-   `fractional-width`: number (defaults to 1)
+-   `min-pixel-width`: number (defaults to minimum supported by table)
+
+_Content_
+
+-   column title (icon and/or text)
+
+_Events_
+
+-   `menu-button-beforetoggle`: Fired when a menu button in a cell fires the `beforetoggle` event
+-   `menu-button-toggle`: Fired when a menu button in a cell fires the `toggle` event
+
+The detail for each of these events will be as follows:
+
+```ts
+interface MenuButtonColumnToggleEventDetail {
+    recordId: string;
+    newState: boolean;
+    oldState: boolean;
+    originalEvent: CustomEvent<MenuButtonToggleEventDetail>;
+}
+```
+
+#### Cell View Component
+
+_Component Name_
+
+-   `nimble-table-column-menu-button-cell-view`
+
+_Rendering_
+
+If the record value associated with the specified `field-name` is a non-empty string, the cell will render a menu button containing the record value text. If the record value associated with the specified `field-name` is not a non-empty string, the cell will render nothing.
+
+The menu button will have the following behaviors/styles:
+
+-   Use `ghost` appearance mode
+-   Have `nimble-icon-arrow-expander-down` slotted into the `end` slot
+-   Grow to fill the width of the cell
+-   Left-align the text (with padding) in the button to make the text within each button is aligned for a given column
+    -   By default, button text is centered, so custom styling will be written to handle this case.
+-   Ellipsize text in the button if it doesn't fit in the button
+-   Provide the full text within the button's `title` if the text is ellipsized
+    -   It is unlikely this can leverage the existing `overflow` directive as-is because the overflow will occur on the text within a `span` within the button, but the `title` should be added when the button is hovered. The best solution to this problem will be determined during implementation.
+
+#### Group Header View Component
+
+A new component will not be created for the group header view of the menu button column. The column will specify the existing `tableColumnTextGroupHeaderViewTag` as the `groupHeaderViewTag` because the header will contain only text.
+
+### Sorting & Grouping
+
+The column will be sorted and grouped based on the text that is provided for display in the button.
```

Comment:
It would be trivial to not have grouping supported in this column (i.e. don't use the groupable column mixin for this column), but I think it would be more difficult to not have sorting for the column. The properties/attributes related to sorting (`sortDirection`, `sortIndex`, and `sortingDisabled`) are all defined on the base `TableColumn` class. Therefore, unless I'm missing something, I think we'd either need to move those into a mixin that every existing column uses, or we'd need to leave those on the menu button column but have the values be ignored (which seems odd). Thoughts?

Reply:
Oh interesting. Yea guess we added all the sorting before we had the mixin pattern? I don't really remember at this point if there was an explicit decision to make grouping opt-in for columns but sorting not. If there isn't a fundamental reason then seems like sorting should be opt-in per column?

Wouldn't think it needs to block this implementation, could document it as not supported / internal for this column and then have creating the mixin as separate work?

Reply:
My hunch is that it wouldn't be too much work to create a mixin for those sorting properties since we already have a well-established mixin pattern. Assuming that's correct, I don't see a reason we couldn't do that as part of this feature.

Reply:
I've updated the HLD to say that we won't support grouping or sorting for this column. I also mentioned that we will need to create a mixing for sorting in order to disable sorting on this column.